nextcord/http.py

```diff
@@ -151,48 +471,172 @@ def __exit__(
 
 
 class HTTPClient:
-    """Represents an HTTP client sending HTTP requests to the Discord API."""
+    """Represents an HTTP client sending HTTP requests to the Discord API.
+
+    Also, not thread safe.
+
+    Parameters
+    ----------
+    connector: Optional[:class:`aiohttp.BaseConnector`]
+        The connector object to use for the client session.
+    max_global_requests: :class:`int`
+        Maximum amount of requests per second per authorization.
+
+        Discord by default only allows 50 requests per second, but if your bot has had its maximum increased, then
+        increase this parameter.
+    time_offset: :class:`float`
+        Amount of seconds added to all ratelimit timers for lag compensation.
+
+        Due to latency and Discord servers not perfectly time synced, having no offset can cause 429's to occur even
+        with us following the reported X-RateLimit-Reset-After.
+
+        Increasing will protect from erroneous 429s but will slow bucket resets, lowering max theoretical speed.
+
+        Decreasing will hasten bucket resets and increase max theoretical speed but may cause 429s.
+    default_auth: Optional[:class:`str`]
+        Default string to use in the Authorization header if it's not manually provided.
```

Comment:
dpy only has one bot per http client iirc and has seperate RL implementations for webhooks? 
Why is this not required?